benchmark-ph1ex1/convert.py

```diff
@@ -0,0 +1,472 @@
+    for i in range(len(XS['SP0'])):
+        for dg in XS['SP0'][i]:
+            f.write(' ' + str(dg))
+    f.write('\n')
+    f.close()
+
+    f = open(base + 'REMXS' + '.txt', "w+")
+    f.write(str(T))
+    for i in range(len(XS['SP0'])):
+        scatii = float(XS['SP0'][i, i])
+        toti = float(XS['ST'][i])
+        f.write(' ' + str(toti-scatii))
+    f.write('\n')
+    f.close()
+
+    # MeV/fission: 200 MeV/fission
+    f = open(base + 'KAPPA' + '.txt', "w+")
+    f.write(str(T))
+    for i in range(G):
+        if index == 'fuel':
+            f.write(' 200.0')
+        else:
+            f.write(' 0.0')
+    f.write('\n')
+    f.close()
+
+    # The remaining constants can be all 0
+    data = ['INVV', 'CHID']
+    for param in data:
+        f = open(base + param + '.txt', "w+")
+        f.write(str(T))
+        for i in range(G):
+            f.write(' 0.0')
+        f.write('\n')
+        f.close()
+
+    # 8 is the number of precursor groups
+    data = ['BETA_EFF', 'LAMBDA']
+    for param in data:
+        f = open(base + param + '.txt', "w+")
+        f.write(str(T))
+        for i in range(8):
+            f.write(' 0.0')
+        f.write('\n')
+        f.close()
+
+    print(index + ' done')
+
+
+def homogenize(XS, vi):
+    '''
+    This function homogenizes several materials into one.
+    The input cross sections have a 26 energy group structure.
+
+    Parameters:
+    -----------
+    XS: dictionary
+        parameters to homogenize
+    vi: array of float
+        volume fraction of each material
+    Returns:
+    --------
+    HXS: dictionary
+        homogenized parameters
+    '''
+
+    HXS = {'FLX': [], 'ST': [], 'DIFFCOEF': [], 'NSF': [], 'FISS': [],
+           'CHIT': [], 'SP0': np.zeros((26, 26))}
+
+    L = len(XS)
+    for group in range(26):
+        summ = 0
+        for material in range(L):
+            summ += float(XS[material]['FLX'][group]) * vi[material]
+        HXS['FLX'].append(summ)
+
+        data = ['ST', 'DIFFCOEF', 'FISS', 'NSF']
+        for dat in data:
+            sumx = 0
+            for material in range(L):
+                sumx += float(XS[material][dat][group])\
+                        * float(XS[material]['FLX'][group]) * vi[material]
+            HXS[dat].append(sumx/summ)
+
+        sumxx = 0
+        for material in range(L):
+            sumxx += float(XS[material]['CHIT'][group])\
+                     * float(XS[material]['NSF'][group])\
+                     * float(XS[material]['FLX'][group])\
+                     * vi[material]
+        if sumx != 0:
+            HXS['CHIT'].append(sumxx/sumx)
+        else:
+            HXS['CHIT'].append(0)
+
+        for gp in range(26):
+            sumx = 0
+            for material in range(L):
+                sumx += float(XS[material]['SP0'][group, gp])\
+                        * float(XS[material]['FLX'][group])*vi[material]
+            HXS['SP0'][group, gp] = sumx/summ
+
+    return HXS
+
+
+def collapse(XS, lim):
```

Comment:
This function has been copy pasted and then modified from another file in this PR (or maybe this was the original, it's impossible to know... ). This is a huge violation of DRY.

Reply:
So far, this is the only ```collapse(XS, lim)``` in the repo. Proably, the other PRs violate DRY, but let's focus on this PR for the moment, and I will fix the other PRs.